src/resolutions/lesson-02/3-infer-props.ts

```diff
@@ -0,0 +1,38 @@
+import { Equal, Expect } from "../../../helper";
+import { expect, it} from "vitest";
+// 3. infer props on clonedComponent :?
+
+type TProps = {
+    a: number;
+    b: number;
+    c: number;
+}
+
+
+export class Component<TProps> {
+  private props: TProps;
+
+  constructor(props: TProps) {
+    this.props = props;
+  }
+
+  getProps = () => this.props;
+}
+
+const cloneComponent = (component: Component<TProps>) => {
+  return new Component(component.getProps());
+};
+
```

Comment:
no deberíamos usar un tipo alias para esto, recuerda que estamos trabajando con genéricos. 
la idea es capturar el genérico en nuestra función `cloneComponent` y hacer uso del.

Reply:
Thx, addressed in next commit.